fetch.bs

```diff
@@ -6710,6 +6748,362 @@ agent's <a>CORS-preflight cache</a> for which there is a <a>cache entry match</a
+       <var>deferredRecord</var>'s <a for="deferred fetch record">request</a>.
+
+       <li><p>Decrement <var>quota</var> by <var>requestLength</var>.
+
+       <li><p>If <var>deferredRecord</var>'s <a for="deferred fetch record">request</a>'s
+       <a for=request>URL</a>'s <a for=url>origin</a> is <a>same origin</a> with <var>origin</var>,
+       then decrement <var>quotaForOrigin</var> by <var>requestLength</var>.
+      </ol>
+    </ol>
+
+   <li>
+    <p>Otherwise, if <var>otherNavigable</var> is not a <a for=/>top-level traversable</a>, and its
+    <a for=navigable>parent</a>'s <a>active document</a> and <var>document</var>
+    <a>share deferred-fetch quota</a>, then decrement <var>quota</var> by
+    <var>otherNavigable</var>'s <a>navigable container</a>'s
+    <a>subframe reserved deferred-fetch quota</a>.
+
+    <p class=note>This is a cross-origin/cross-agent subframe, so it spends some of the quota of
+    this document and its relatives.
+  </ol>
+ <li><p>If <var>quota</var> is less than zero, than return zero.
+
+ <li><p>If <var>quota</var> is less than <var>quotaForOrigin</var>, then
+ return <var>quota</var>.
+
+ <li><p>Return <var>quotaForOrigin</var>.
+</ol>
+</div>
+
+<div algorithm>
+<p>To <dfn export>reserve deferred-fetch quota</dfn> for a <a>navigable container</a>
+<var>container</var> given an <a for=/>origin</a> <var>originToNavigateTo</var>:
+
+<p class=note>This is called when the container document initiates a navigation. It potentially
+reserves either 64kb or 8kb of quota for the frame, if it doesn't <a>share deferred-fetch quota</a>
+with the container and the permissions policy allow. It is not observable to the cotnainer document
+whether the reserved quota was used in practice.
+This algorithm assumes that the container's document might delegate quota to the navigated frame,
+and the reserved quota would only apply in that case, and would be ignored if it ends up being
+shared.
+
+<ol>
+ <li><p>If the <a data-lt="define an inherited policy for feature in container">inherited policy</a>
+ for {{PermissionPolicy/"deferred-fetch"}}, <var>container</var> and <var>originToNavigateTo</var>
+ is <code>Enabled</code>, and the <a>available deferred-fetch quota</a> for
+ <var>container</var>'s <a>container document</a> is equal or greater than
+ <a>optional subframe deferred-fetch quota</a>, then set <var>container</var>'s
+ <a>reserved deferred-fetch quota</a> to <a>optional subframe deferred-fetch quota</a>.
+
+ <li>
+  <p>Otherwise, if the
+  <a data-lt="define an inherited policy for feature in container">inherited policy</a> for
+  {{PermissionPolicy/"deferred-fetch-minimal"}}, <var>container</var> and
+  <var>originToNavigateTo</var> is <code>Enabled</code>, then:
+  <ol>
+   <li><p>Let <var>delegatedQuota</var> be <a>deferred-fetch delegated quota</a>.
+
+   <li>
+    <p><a for=list>For each</a> <var>navigable</var> of <var>container</var>'s
```

Comment:
What's the result of this for-loop given navigable=`frame-3`/`frame-4` from the following example? Is it `{frame-3, frame-4}`?

```
<iframe src="frame-1.html"></iframe>
<iframe src="frame-2.html"></iframe>
<iframe src="http://b.com/frame-3.html"></iframe>
<iframe src="http://b.com/frame-4.html"></iframe>
```

Reply:
It would be `[frame-3]` if navigable is `frame-4`, and `[frame-4]` if navigable is `frame-3`, because of the first condition: "`navigable` is not `container`s `content navigable`".